Objects/abstract.c

```diff
@@ -2881,7 +2881,47 @@ PyAIter_Check(PyObject *obj)
             tp->tp_as_async->am_anext != &_PyObject_NextNotImplemented);
 }
 
+static int
+iternext(PyObject *iter, PyObject **item)
+{
+    *item = (*Py_TYPE(iter)->tp_iternext)(iter);
+    if (*item == NULL) {
+        PyThreadState *tstate = _PyThreadState_GET();
+        /* When the iterator is exhausted it must return NULL;
+         * a StopIteration exception may or may not be set. */
+        if (!_PyErr_Occurred(tstate)) {
+            return 0;
+        }
+        if (_PyErr_ExceptionMatches(tstate, PyExc_StopIteration)) {
+            _PyErr_Clear(tstate);
+            return 0;
+        }
+        return -1;
+    }
+    return 1;
+}
+
+/* Return 1 and set 'item' to the next item of iter on success.
+ * Return 0 and set 'item' to NULL when there are no remaining values.
+ * Return -1, set 'item' to NULL and set an exception on error.
+ */
+int
+PyIter_NextItem(PyObject *iter, PyObject **item)
+{
+    assert(iter != NULL);
+    assert(item != NULL);
+
+    if (!PyIter_Check(iter) && !PyAIter_Check(iter)) {
+        PyErr_Format(PyExc_TypeError, "expected an iterator, not '%T'", iter);
+        return -1;
+    }
+
+    return iternext(iter, item);
+}
+
 /* Return next item.
+ * Deprecated; use PyIter_NextItem() instead.
```

Comment:
I would mark add this comment in abstract.h as well because my IDE jumps to the declaration, not to the definition (and in general I write comments at the declaration but not the definition level, but that's my personal opinion).